### PR TITLE
feat(client): reduce generated HTTP request header code size

### PR DESCRIPTION
**Pull request by AndrewFossAWS**
### Description
This PR reduces generated HTTP request header code size.
* Bump Smithy version to 1.30.0
* See PR for smithy-ts changes https://github.com/awslabs/smithy-typescript/pull/729
* See code snippet below for generated code changes:

Before:
```
  const headers: __HeaderBag = {
    "content-type": "application/x-amz-json-1.1",
    "x-amz-target": "ACMPrivateCA.CreateCertificateAuthority",
  };
```

After:
(With operation name)
```
function sharedHeaders(operation: string): __HeaderBag {
  return {
    "content-type": "application/x-amz-json-1.1",
    "x-amz-target": "ACMPrivateCA.${operation}",
  };
}

const headers: __HeaderBag = sharedHeaders("CreateCertificateAuthority");
```
(Without operation name)
```
const SHARED_HEADERS: __HeaderBag = {
  "content-type": "application/x-www-form-urlencoded",
};

const headers: __HeaderBag = SHARED_HEADERS;
```

### Testing
yarn test:all

### Additional context
Add any other context about the PR here.

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: codegen/gradle.properties [hunk 1]
```diff
@@ -1,2 +1,3 @@
 smithyVersion=1.28.1
-smithyGradleVersion=0.6.0
\ No newline at end of file
+smithyGradleVersion=0.6.0
+org.gradle.jvmargs=-Xmx1024m
```
Comment (srchase) [inline codegen/gradle.properties:3]: Was this change intended to be included in the PR? It isn't called out in the description and seems unrelated.

File: codegen/gradle.properties [hunk 2]
```diff
@@ -1,2 +1,2 @@
-smithyVersion=1.28.1
+smithyVersion=1.30.0
```
Comment (srchase) [inline codegen/gradle.properties:1]: I think it'd be helpful to mention in the description that this is bumping the Smithy version to 1.30.0.

**Review by srchase — APPROVED** [review]

**Review by srchase — APPROVED** [review]

**Review by kuhe — APPROVED** [review]
